__tests__/integration/app.test.js

```diff
@@ -0,0 +1,56 @@
+const app = require("../../app");
+const request = require("supertest");
+const fact = require("../../models/fact");
+const api = require("../../routes/api");
+
+describe("Integration/unit testing app.js", function () {
+  describe("GET /", function () {
+    test("GET request responds with a 200 status code and returns HTML", async function (done) {
+      const response = await request(app).get("/");
+      const { statusCode, text } = response;
+      expect(statusCode).toBe(200);
+      expect(text).toContain("<!DOCTYPE html>");
+      done();
```

Comment:
hmm, do you need this `done` pattern? i'd try removing it and seeing if the tests still work.

Reply:
(this comment applies to all tests in this suite)

Reply:
removing the done() callbacks gives this error message for each of the tests - 
`    : Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Error:
`

Reply:
@mmmaaatttttt removing the `done()` pattern causes the tests to fail. I'm not sure there's another way to fix it without manually configuring jest.

Reply:
Hey @stephenisau - I can take out these `done` references and everything works fine. What version of node are you using? I'm on 12.2.0.

Reply:
Somewhat relatedly, I'm not sure if this is coming from the `app` tests or the `data` tests, but when I run the tests, my console gets hammered by warnings and errors. can you please make sure that both the tests pass, and the output doesn't contain warnings / errors? you should be able to mock console.warn and error and replace them with implementations that don't actually log anything. see here for some ideas: https://medium.com/swlh/6-ways-to-run-jest-test-cases-silently-67d2fead8c11

Reply:
yeppers, i've added some jest configurations and a `setup.js` file that runs before each test suite. Referenced the article you've provided and left it as a comment in the `setup.js` file!